test/core/integration/concrete/lockup/burn/burn.t.sol

```diff
@@ -2,177 +2,142 @@
         vm.warp({ newTimestamp: defaults.CLIFF_TIME() });
         resetPrank({ msgSender: users.sender });
-        lockup.cancel(streamId);
+        lockup.cancel(defaultStreamId);
         resetPrank({ msgSender: users.recipient });
-        vm.expectRevert(abi.encodeWithSelector(Errors.SablierLockup_StreamNotDepleted.selector, streamId));
-        lockup.burn(streamId);
-    }
-
-    modifier givenDepletedStream(uint256 streamId_) {
-        vm.warp({ newTimestamp: defaults.END_TIME() });
-        lockup.withdrawMax({ streamId: streamId_, to: users.recipient });
-        _;
+        vm.expectRevert(abi.encodeWithSelector(Errors.SablierLockup_StreamNotDepleted.selector, defaultStreamId));
+        lockup.burn(defaultStreamId);
     }
 
-    function test_RevertWhen_UnauthorizedCaller()
+    function test_RevertWhen_CallerMaliciousThirdParty()
         external
         whenNoDelegateCall
         givenNotNull
-        givenDepletedStream(streamId)
+        givenDepletedStream(lockup, defaultStreamId)
+        whenCallerNotRecipient
     {
         resetPrank({ msgSender: users.eve });
-        vm.expectRevert(abi.encodeWithSelector(Errors.SablierLockup_Unauthorized.selector, streamId, users.eve));
-        lockup.burn(streamId);
-    }
-
-    modifier whenAuthorizedCaller() {
-        _;
+        vm.expectRevert(abi.encodeWithSelector(Errors.SablierLockup_Unauthorized.selector, defaultStreamId, users.eve));
+        lockup.burn(defaultStreamId);
     }
 
-    function test_RevertGiven_NFTNotExist()
+    function test_RevertWhen_CallerSender()
         external
         whenNoDelegateCall
         givenNotNull
-        givenDepletedStream(streamId)
-        whenAuthorizedCaller
+        givenDepletedStream(lockup, defaultStreamId)
+        whenCallerNotRecipient
     {
-        // Burn the NFT so that it no longer exists.
-        lockup.burn(streamId);
-
-        // Run the test.
-        vm.expectRevert(abi.encodeWithSelector(IERC721Errors.ERC721NonexistentToken.selector, streamId));
-        lockup.burn(streamId);
-    }
-
-    modifier givenNFTExists() {
-        _;
+        resetPrank({ msgSender: users.sender });
+        vm.expectRevert(
+            abi.encodeWithSelector(Errors.SablierLockup_Unauthorized.selector, defaultStreamId, users.sender)
+        );
+        lockup.burn(defaultStreamId);
     }
 
-    function test_GivenNonTransferableNFT()
+    function test_WhenCallerApprovedThirdParty()
         external
         whenNoDelegateCall
         givenNotNull
-        givenDepletedStream(notTransferableStreamId)
-        whenAuthorizedCaller
-        givenNFTExists
+        givenDepletedStream(lockup, defaultStreamId)
+        whenCallerNotRecipient
     {
-        // It should emit a {MetadataUpdate} event.
-        vm.expectEmit({ emitter: address(lockup) });
-        emit MetadataUpdate({ _tokenId: notTransferableStreamId });
+        resetPrank({ msgSender: users.recipient });
```

Comment:
Shouldn't this be `operator`?

Reply:
Since we have `users.operator`, we should do this:

- In `Integration_Test`, `recipient` should call `setApprovalForAll` on `users.operator`. This will make operator an approved third party for all the tokens.
- Then we should prank to `users.operator` wherever we have `test_WhenCallerApprovedThirdParty` in the test. If we prank to alice there, then why do we have to operator in the first place.

Wdyt?

Reply:
we can't do this in `Integration_Test` as the LD/LL/LT are not yet deployed there, it should be

it should be in `LockupDynamic_Integration_Shared_Test` `LockupLinear_Integration_Shared_Test` `LockupTranched_Integration_Shared_Test` if we want to do this

Reply:
Why cant we do it in `Integration_Test.setUp()` or `Base_Test.setUp`? The lockup contracts are deployed in Base setup through `deployProtocolConditionally` so I wonder why we cant do this.

Reply:
This is what I was talking about: https://github.com/sablier-labs/v2-core/pull/1058/commits/7cc0f9dbbeb5f2608c7440b572769339759e0433

Reply:
yes it looks good, i was trying to use `lockup.setApprovalForAll` which doesn't work

Reply:
but was there problem in keeping them in Lockup<Model>_Integration_Shared_Test

Reply:
Yes there is a small problem which is since we initialize all users including `operator` in the Base contract, `setApprovalForAll` permission is a part of setting up `operator` user. The operator is not an operator until `setApprovalForAll` is called by the recipient. Therefore, its better to have it next to users declaration than in the `Lockup<Model>_Integration_Shared_Test`.